pkg/config/nodetreemodel/config.go

```diff
@@ -296,6 +298,24 @@ func (c *ntmConfig) mergeAllLayers() error {
 	return nil
 }
 
+func computeAllSettings(node InnerNode, path string) []string {
+	knownKeys := []string{}
+	for _, name := range node.ChildrenKeys() {
+		newPath := name
+		if path != "" {
+			newPath = joinKey(path, name)
```

Comment:
What do you think about filtering empty strings inside `joinKey` instead? To me it feels like it would be more reusable.